JacksonDunstanNativeCollections/NativePerJobThreadIntPtrs.cs

```diff
@@ -0,0 +1,434 @@
+		public Parallel GetParallel()
+		{
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+			Parallel parallel = new Parallel(Length, m_Buffer, m_Safety);
+			AtomicSafetyHandle.UseSecondaryVersion(ref parallel.m_Safety);
+#else
+			Parallel parallel = new Parallel(m_Buffer);
+#endif
+			return parallel;
+		}
+
+		/// <summary>
+		/// Check if the underlying unmanaged memory has been created and not
+		/// freed via a call to <see cref="Dispose"/>.
+		/// 
+		/// This operation has no access requirements.
+		///
+		/// This operation is O(1).
+		/// </summary>
+		/// 
+		/// <value>
+		/// Initially true when a non-default constructor is called but
+		/// initially false when the default constructor is used. After
+		/// <see cref="Dispose"/> is called, this becomes false. Note that
+		/// calling <see cref="Dispose"/> on one copy of this object doesn't
+		/// result in this becoming false for all copies if it was true before.
+		/// This property should <i>not</i> be used to check whether the object
+		/// is usable, only to check whether it was <i>ever</i> usable.
+		/// </value>
+		public bool IsCreated
+		{
+			get
+			{
+				return m_Buffer != null;
+			}
+		}
+
+		/// <summary>
+		/// Release the object's unmanaged memory. Do not use it after this. Do
+		/// not call <see cref="Dispose"/> on copies of the object either.
+		/// 
+		/// This operation requires write access.
+		/// 
+		/// This complexity of this operation is O(1) plus the allocator's
+		/// deallocation complexity.
+		/// </summary>
+		[WriteAccessRequired]
+		public void Dispose()
+		{
+			RequireWriteAccess();
+
+// Make sure we're not double-disposing
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+#if UNITY_2018_3_OR_NEWER
+        	DisposeSentinel.Dispose(ref m_Safety, ref m_DisposeSentinel);
+#else
+			DisposeSentinel.Dispose(m_Safety, ref m_DisposeSentinel);
+#endif
+#endif
+
+			UnsafeUtility.Free(m_Buffer, m_AllocatorLabel);
+			m_Buffer = null;
+		}
+
+		/// <summary>
+		/// Set whether both read and write access should be allowed. This is
+		/// used for automated testing purposes only.
+		/// </summary>
+		/// 
+		/// <param name="allowReadOrWriteAccess">
+		/// If both read and write access should be allowed
+		/// </param>
+		[Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
+		[BurstDiscard]
+		public void TestUseOnlySetAllowReadAndWriteAccess(
+			bool allowReadOrWriteAccess)
+		{
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+			AtomicSafetyHandle.SetAllowReadOrWriteAccess(
+				m_Safety,
+				allowReadOrWriteAccess);
+#endif
+		}
+
+		/// <summary>
+		/// Throw an exception if the object isn't readable
+		/// </summary>
+		[Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
+		[BurstDiscard]
+		private void RequireReadAccess()
+		{
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+			AtomicSafetyHandle.CheckReadAndThrow(m_Safety);
+#endif
+		}
+
+		/// <summary>
+		/// Throw an exception if the object isn't writable
+		/// </summary>
+		[Conditional("ENABLE_UNITY_COLLECTIONS_CHECKS")]
+		[BurstDiscard]
```

Comment:
This shouldn't have `[BurstDiscard]` because we want the checks when Burst is enabled with safety checks.